website/blog/apr-2023/silk2170.md

```diff
@@ -0,0 +1,167 @@
+
+### Behind the Scenes
+
+To improve the usability of our DirectX family of bindings, we took several steps in Silk.NET 2.17. First, we added extension methods to the ComPtr<T> type for each interface struct within the bindings. While this gave users the ability to call methods directly on the ComPtr instance, it didn't do enough to improve usability on its own.
+
+To address this, we introduced two new overloaders: the ComPtrOverloader and the NonKhrReturnTypeOverloader. The ComPtrOverloader looks for parameter signatures that match the IID_PPV_ARGS pattern, and creates a generic ref ComPtr<T> overload coalescing those two parameters. The interface ID is implicitly retrieved from the interface struct used as the generic type argument to ComPtr.
+
+The NonKhrReturnTypeOverloader takes things a step further by identifying functions whose last argument is a ref (which, after the ComPtrOverloader has worked its magic, will be the case for methods that return COM pointers). It changes that parameter to be the return type of that function instead. This is what gives users the nice CreateDXGIFactory2 experience seen in the example snippet earlier in this post.
+
+We're always trying to improve the experience of using our bindings, so please let us know in [this tracking issue](https://github.com/dotnet/Silk.NET/issues/887) if you have any feedback or ideas for improvements! As always, you can also reach us in Discord or GitHub Discussions.
+
+## WebGPU bindings
+
+Development Lead: Beyley Thomas (@Beyley)
+
+We've added bindings to the experimental [webgpu-headers](https://github.com/webgpu-native/webgpu-headers) maintained by the various vendors of WebGPU drivers! WebGPU is an API that exposes the capabilities of GPU hardware for the Web. The API is designed from the ground up to efficiently map to modern native GPU APIs such as Vulkan and Direct3D 12, and is touted to be the modern replacement for WebGL.
+
+While these bindings have "web" in the name, using either the Dawn or WGPU implementations of WebGPU (for which we ship native binaries) you can use this as a single API to target all of the various platforms that would otherwise have platform-specific APIs! This allows you to write truly write-once-run-everywhere code.
+
+<?# FancyImage "../../images/blog/apr-2023/wgpuquad.jpg" "The Silk.NET logo rendered using WebGPU." "The Silk.NET logo rendered using WebGPU." /?>
+
+Please note these bindings do not work in browsers yet due to issues with the .NET WebAssembly runtime.
+
+## More DirectX Bindings
+
+Development Lead: Andrew Davis (@curin) \[Silk.NET Community\]
+
+In addition to the major improvements in usability for our DirectX family of bindings, we're excited to announce that Silk.NET 2.17 also includes bindings for Direct2D, DirectComposition, DirectWrite, and WIC. Direct2D has been among the most requested features from our users, and we're thrilled to be able to deliver on them with this release.
```

Comment:
we do not have WIC rn, unless thats a "lets add before 2.17 rq thing" although i suspect it may be a pain and take awhile

Reply:
Removed